lib/geordi/gitpt.rb

```diff
@@ -96,4 +119,39 @@ def bold(string)
     HighLine::BOLD + string + HighLine::RESET
   end
 
+  def checkout_branch(name, new_branch: false)
+    if new_branch
+      Geordi::Util.system! 'git', 'checkout', 'master'
+      Geordi::Util.system! 'git', 'checkout', '-b', name
+    else
+      Geordi::Util.system! 'git', 'checkout', name
+    end
+  end
+
+  def normalize_branch_name(name)
+    name.gsub!('ä', 'ae')
+    name.gsub!('ö', 'oe')
+    name.gsub!('ü', 'ue')
+    name.gsub!('ß', 'ss')
+    name.tr!('^A-Za-z0-9_ ', '')
+    name.squeeze! ' '
+    name.gsub!('-', '_')
+    name.gsub!(' ', '_')
+    name.downcase!
+    name
+  end
+
+  def git_user_initials
+    stdout_str, _error_str, status = Open3.capture3('git', 'config', 'user.name')
+    git_user_initials = unless stdout_str.nil?
+      stdout_str.strip.split(' ').map(&:chars).map(&:first).join.downcase
+    end
+
+    if !status.success? || git_user_initials.nil?
+      Interaction.fail('Could not determine the git users initals')
+    else
+      git_user_initials
```

Comment:
Please verify the initials by prompting the user (with your derived initials as default value). See https://github.com/makandra/geordi/blob/c67008023b64dd88cbbd1bed0ddb3a7b64d97e58/lib/geordi/interaction.rb#L47-L49 and https://github.com/makandra/geordi/blob/c67008023b64dd88cbbd1bed0ddb3a7b64d97e58/lib/geordi/commands/deploy.rb#L60

I know of a colleague that cannot use his "natural initials" because of namespace collision. He needs a chance to enter custom initials.

Reply:
Bonus: asking for the initials makes failing if blank in L151 obsolete—you'll simply prompt without default value.